### ISSUE TITLE
Clean up scheduled timer messages when timers are cancelled

### ISSUE THREAD

**Issue by cschleiden** [cschleiden/go-workflows#19]

**Comment by cschleiden** [comment]
Investigate what should happen with pending timer events if a workflow ends before the timer is fired. For example, when in a race between a timer and an activity an activity wins.

**Comment by cschleiden** [comment]
DurableTask doesn't mark an orchestration complete until all open tasks have a result:

```csharp
// check if workflow is completed after this replay
                    if (!this.context.HasOpenTasks)
```

**Comment by cschleiden** [comment]
Related: https://github.com/Azure/durabletask/issues/415

Think it might be best to have a workflow finish anyway, clean up any outstanding messages and potentially issue a warning if there are any.

**Comment by cschleiden** [comment]
This is implemented for the Redis backend.

**Comment by cschleiden** [comment]
This has been fixed in #94
